src/main/java/fi/hsl/common/monitoring/MonitoringScheduler.java

```diff
@@ -0,0 +1,81 @@
+package fi.hsl.common.monitoring;
+
+import com.typesafe.config.Config;
+import fi.hsl.common.monitoring.proto.Monitoring;
+import fi.hsl.common.pulsar.PulsarApplicationContext;
+import org.apache.pulsar.client.api.Producer;
+import org.apache.pulsar.client.api.TypedMessageBuilder;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class MonitoringScheduler {
+    protected static final Logger LOGGER = LoggerFactory.getLogger(MonitoringScheduler.class);
+
+    protected static List<AbstractMonitoringHandler> handlers = new ArrayList<>();
+    protected static ScheduledExecutorService scheduler = Executors.newSingleThreadScheduledExecutor();;
+    protected static long delay;
+    protected static long interval;
+    protected static Producer producer;
+    protected static Runnable runnable;
+
+    protected static MonitoringScheduler ourInstance = new MonitoringScheduler();
+
+    public static MonitoringScheduler getInstance(final PulsarApplicationContext context, final long delay, final long interval, final Runnable runnable) {
+        producer = context.getProducer();
```

Comment:
Bug in the Singleton pattern here, if you call getInstance() twice with different contexts, the static producer will get overridden. Same goes for runnable. "Real" singleton pattern would store the instance into static member and then return that if once defined. One option is to make the member fields non-static, then calling getInstance is safe. 

Dunno if this is a bug or just a feature here?

Reply:
Indeed, perhaps this shouldn't be singleton.